test/Polly.Specs/Timeout/TimeoutAsyncSpecs.cs

```diff
@@ -248,6 +250,31 @@ public async Task Should_rethrow_exception_from_inside_delegate__pessimistic()
         await policy.Awaiting(p => p.ExecuteAsync(() => throw new NotImplementedException())).Should().ThrowAsync<NotImplementedException>();
     }
 
+    [Fact]
+    public async Task Should_cancel_downstream_token_on_timeout__pessimistic()
+    {
+        // It seems that there's a difference in the mocked clock vs. the real time clock.
+        // This test does not fail when using the mocked timer.
+        // In the TimeoutSpecsBase we actually cancel the combined token. Which hides
+        // the fact that it doesn't actually cancel irl.
+        SystemClock.Reset();
+
+        var policy = Policy.TimeoutAsync(TimeSpan.FromMilliseconds(200), TimeoutStrategy.Pessimistic);
+        var upstreamToken = new CancellationToken();
+        bool isCancelled = false;
+
+        var act = () => policy.ExecuteAsync(async (combinedToken) =>
+        {
+            combinedToken.Register(() => isCancelled = true);
+            await SystemClock.SleepAsync(TimeSpan.FromMilliseconds(300), combinedToken);
+        }, upstreamToken);
+
+        await act.Should().ThrowAsync<TimeoutRejectedException>();
+
+        await SystemClock.SleepAsync(TimeSpan.FromMilliseconds(400), default);
```

Comment:
Maybe make these 300/400ms sleeps longer so there's less chance of a slow CI system causing the test to pass unexpectedly or otherwise introduce flakiness?

Reply:
I think having just one test that's slow because it needs to be to reproduce the issue is fine. A second should be fine as then it's 5x longer than the shortest one.

Reply:
Okay I made the sleep inside 1000ms, and left the cancellation 200 ms. I removed the last wait, because it was unnecessary and could actually hide a race condition. I had it there to prove however long you waited it would never succeed. But I think now that it's fixed this way, to avoid regression we actually don't want it.